File: src/main/java/com/hyperdondon/blissgemsremake/api/util/VectorList.java
package com.hyperdondon.blissgemsremake.api.util;

import java.util.HashMap;
import java.util.Set;

public class VectorList<K, V1, V2> {
    public final HashMap<K, V1> VALUE_1_LIST = new HashMap<>();
    public final HashMap<K, V2> VALUE_2_LIST = new HashMap<>();

    public VectorList() {
    }

    public void add(K key, V1 value1, V2 value2) {
        VALUE_1_LIST.put(key, value1);
        VALUE_2_LIST.put(key, value2);
    }

    public void remove(K key) {
        VALUE_1_LIST.remove(key);
        VALUE_2_LIST.remove(key);
    }

    public V1 getValue1(K key) {
        return VALUE_1_LIST.get(key);
    }

    public V2 getValue2(K key) {
        return VALUE_2_LIST.get(key);
    }

    public void setValue1(K key, V1 value) {
        VALUE_1_LIST.replace(key, value);
    }

    public void setValue2(K key, V2 value) {
        VALUE_2_LIST.replace(key, value);
    }

    public boolean containsKey(K key) {
        return VALUE_1_LIST.containsKey(key);
    }

    public Set<K> keySet() {
        return VALUE_1_LIST.keySet();
    }
}

File: src/main/kotlin/com/hyperdondon/blissgemsremake/internal/util/VectorList.kt
package com.hyperdondon.blissgemsremake.internal.util

class VectorList<K, V1, V2> {
    val value1list: HashMap<K?, V1?> = HashMap()
    val value2list: HashMap<K?, V2?> = HashMap()

    fun add(key: K?, value1: V1?, value2: V2?) {
        value1list.put(key, value1)
        value2list.put(key, value2)
    }

    fun remove(key: K?) {
        value1list.remove(key)
        value2list.remove(key)
    }

    fun getValue1(key: K?): V1? {
        return value1list.get(key)
    }

    fun getValue2(key: K?): V2? {
        return value2list.get(key)
    }

    fun setValue1(key: K?, value: V1?) {
        value1list.replace(key, value)
    }

    fun setValue2(key: K?, value: V2?) {
        value2list.replace(key, value)
    }

    fun containsKey(key: K?): Boolean {
        return value1list.containsKey(key)
    }

    fun keySet(): MutableSet<K?> {
        return value1list.keys
    }
}
